### PR TITLE
ENH: Draft JOSS paper

### PR DESCRIPTION
**Pull request by jvkersch**
Happy new year!

This is a draft for a [JOSS](https://joss.theoj.org/) submission ﻿about vpsearch. I think the paper is submission ready, modulo perhaps some tweaks.

You can get a preview of what the paper will look like at https://whedon.theoj.org/.

### CODE REVIEW

**Review by stevenjkern — CHANGES_REQUESTED** [review]
Looks lovely, @jvkersch! I have just a few very small comments. 🙏🏻 ❤️

File: paper/paper.md [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+tree_ [@1991-uhlmann-SatisfyingGeneralProximity;
+@1993-yianilos-DataStructuresAlgorithms], a data structure that allows for
+similarity lookups in time proportional to the logarithm of the size of the
+database. For a set of unknown sequences, VPsearch is then able to query this
+tree and return the best matching results from the database. In the case study
+below we show that for short sequences (such as the 16S rRNA gene used in
+bacterial classification) VPsearch outperforms both BLAST (7x speedup) and
+ggsearch36 from the FASTA suite (27x speedup) without any loss in accuracy.
+
+The VPsearch tool is implemented in Python, using Cython for
```
[the remark this answers is not in the document]
Comment (jvkersch) [reply paper/paper.md:70]: Good idea. There's an [IEEE paper ](https://ieeexplore.ieee.org/document/5582062)(from 2011) that is mentioned on the Cython homepage, I'll add that.

**Comment by jvkersch** [comment]
@stevenjkern Thanks for picking up on this and for the quick turnaround in reviewing. I was going to send you an email about this, but you saw the PR before I had done so.

I think I've addressed your comments, let me know if this looks good. If you approve of the PR, I'll take that as a sign that the paper is ready for submission to JOSS.

File: paper/paper.bib [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+  pages = {93--118},
+  issn = {2574-3414, 2574-3414},
+  doi = {10.1146/annurev-biodatasci-072018-021156},
+  abstract = {Large-scale genomics demands computational methods that scale sublinearly with the growth of data. We review several data structures and sketching techniques that have been used in genomic analysis methods. Specifically, we focus on four key ideas that take different approaches to achieve sublinear space usage and processing time: compressed full text indices, approximate membership query data structures, locality-sensitive hashing, and minimizers schemes. We describe these techniques at a high level and give several representative applications of each.},
+  langid = {english}
+}
+
+@ARTICLE{2011-behnel-cython,
+  author={Behnel, Stefan and Bradshaw, Robert and Citro, Craig and Dalcin, Lisandro and Seljebotn, Dag Sverre and Smith, Kurt},
+  journal={Computing in Science   Engineering},
```
Comment (stevenjkern) [inline paper/paper.bib:159]:
```suggestion
  journal={Computing in Science \& Engineering},
```

**Review by stevenjkern — APPROVED** [review]
LGTM modulo missing &

:shipit:

**Comment by jvkersch** [comment]
Thanks @stevenjkern, good catch! I'll merge the PR now and submit off the draft, and we can open separate PRs for follow-up changes.

**Pull request merged**